packages/next/src/server/app-render/collect-segment-data.tsx

```diff
@@ -19,9 +19,43 @@ import { UNDERSCORE_NOT_FOUND_ROUTE } from '../../api/constants'
 import { waitAtLeastOneReactRenderTask } from '../../lib/scheduler'
 import type { LoadingModuleData } from '../../shared/lib/app-router-context.shared-runtime'
 
+// Contains metadata about the route tree. The client must fetch this before
+// it can fetch any actual segment data.
+type RootTreePrefetch = {
+  tree: TreePrefetch
+  staleTime: number
+}
+
+type TreePrefetch = {
+  // The key to use when requesting the data for this segment (analogous to a
+  // URL). Also used as a cache key, although the server may specify a different
+  // cache key when it responds (analagous to a Vary header), like to omit
+  // params if they aren't used to compute the response. (This part not
+  // yet implemented)
+  key: string
+
+  // Child segments.
+  slots: null | {
+    [parallelRouteKey: string]: TreePrefetch
+  }
+
+  // Extra fields that only exist so we can reconstuct a FlightRouterState on
+  // the client. We may be able to unify TreePrefetch and FlightRouterState
+  // after some refactoring, but in the meantime it would be wasteful to add a
+  // bunch of new prefetch-only fields to FlightRouterState. So think of
+  // TreePrefetch as a superset of FlightRouterState.
+  extra: [segment: Segment, url: string | null, isRootLayout: boolean]
```

Comment:
Yeah I tried to find where it was used but didn't see anything. I figure I'll leave it here until we can remove it from FlightRouterState, too, just in case.

Reply:
Yeah, I think it's somewhat similar to `refetch` in that it's something the client adds to the flight router state rather than being produced by the server.

Although in the case of `url`, I think it's exclusively used by the client. It's how we encode which URLs caused a parallel route to become "active" for the refresh case. 

Worth revisiting later for sure

Reply:
oh I see, ok that seems safe to omit then